core/src/main/java/io/questdb/griffin/model/QueryModel.java

```diff
@@ -1142,6 +1142,7 @@ public void removeColumn(int columnIndex) {
         aliasToColumnMap.remove(columnAlias);
         aliasToColumnNameMap.remove(columnAlias);
         columnAliasIndexes.remove(columnAlias);
+        updateColumnAliasIndexes();
```

Comment:
Do you have a reproducer for this?

Reply:
> But, the API allows popping columns from the middle of the range and provides an inconsistent result when you do.

So it was due to a broken iteration order. I agree that the API is unsafe, but we have many of unsafe APIs around. Maybe we should document this behavior to avoid bugs introduced in future?

Reply:
> Imo its reasonable to expect the column maps to be in a consistent state after removing a column, and not require the caller to rebuild the index manually.

That's a fair point. If you have an idea on how to improve the API, go for it.